src/devtools/client/webconsole/selectors/messages.ts

```diff
@@ -35,15 +35,30 @@ export const getVisibleMessages = createSelector(
   getAllMessagesById,
   (state: UIState) => state.messages.visibleMessages,
   getFocusRegion,
-  (messages, visibleMessages, focusRegion) => {
-    const msgs = visibleMessages;
-
-    if (focusRegion) {
-      return msgs.filter(id => {
-        const msg = messages.entities[id]!;
-        return isInTrimSpan(msg.executionPointTime, focusRegion);
-      });
-    }
+  (state: UIState) => state.app.loadedRegions,
+  (messages, visibleMessages, focusRegion, loadedRegions) => {
+    const msgs = visibleMessages.filter(messageId => {
+      const message = messages.entities[messageId]!;
+      const executionPointTime = message.executionPointTime!;
+
+      // Filter out messages that aren't within the focused region.
+      if (focusRegion) {
+        if (!isInTrimSpan(executionPointTime, focusRegion)) {
+          return false;
+        }
+      }
+
+      // Filter out messages that haven't yet been loaded.
+      if (
+        !loadedRegions!.loaded.some(
```

Comment:
Kind of a nit but I think this is the same as saying 

```
return loadedRegions!.loaded.some(region => executionPointTime >= region.begin.time && executionPointTime <= region.end.time);
```

Which it might be nice to pull out into a `isInLoadedRegions` function, since that's a pretty reasonable check that we might use elsewhere. Actually, maybe a better signature would be like `isInRegions(point, regions)`?

Reply:
> Kind of a nit but I think this is the same as saying

Yeah, definitely. I know this sort of thing is totally subjective, but I kind of find the format I used to be a little more readable in terms of the sequence of conditionals that we check.

I don't feel strongly over this though.

🧐

https://github.com/RecordReplay/devtools/blob/1b65b594016980128185df9b7baaf093e5d04e36/src/ui/reducers/app.ts#L359-L363